dev/run.py

```diff
@@ -0,0 +1,272 @@
+        print_if(verbose, 'missing environment variables: {}'.format(missing_env))
+
+        if not (default or interactive):
+            raise ValueError(
+                """Use default or interactive mode to set required environment variables: 
+                {}""".format(",".join(missing_env)))
+
+        if default:
+            print_if(verbose, 'using default values')
+            missing_env = {k: default_env[k] for k in missing_env}
+
+        if interactive:
+            print_if(verbose, 'enter values for the following')
+            print_if(default and verbose, 'if left blank, default values will be used')
+            missing_env = {k: _safe_input_prompt(k, v) for k, v in missing_env.items()}
+
+    return given_env, missing_env
+
+
+def _env2shellcode(env):
+    # Dict[str, str] -> str
+    return "\n".join("export {}={}".format(k, v) for k, v in env.items())
+
+
+def env_setup(default=False, interactive=False, missing_only=False, override=False, configure=False,
+             auto_completion=False, verbose=False):
+    """
+    Prints out shell commands that can be used in terminal to setup the environment.
+
+    This tool inspects the current environment, and/or adds default values, and/or interactively
+    asks user for values and prints all or the missing variables that need to be set. It can also
+    provide completion for this script. You can source the setup as follows:
+
+    ```
+    python/run.py envsetup -d -c > ./python/.setup.sh && source ./python/.setup.sh
+    ```
+
+    :param default: if default values should be set in this script or not
+    :param interactive: if user should be prompted for values or not
+    :param missing_only: if only missing variable should be printed
+    :param override: if current environment variables should be overridden
+    :param configure: if configuration should be printed or not
+    :param auto_completion: if auto complete code should be printed
+    :param verbose: if user should be guided or not
+    """
+    given_env, missing_env = _get_required_env(
+        default=default, interactive=interactive, override=override, verbose=verbose)
+
+    # print according to options
+    output_env = {}
+    output_env.update(missing_env)
+
+    if not missing_only:
+        output_env.update(given_env)
+
+    if configure:
+        required_env = dict(given_env)
+        required_env.update(missing_env)
+        configured_env = _get_configured_env(required_env)
+        output_env.update(configured_env)
+
+    env_str = "#!/bin/bash\n"
+    env_str += _env2shellcode(output_env)
+
+    if auto_completion:
+        env_str += argcomplete.shellcode(sys.argv[0])
+
+    print(env_str)
+    return 0
+
+
+def python_tests(*args):
+    """Wrapper for python/run-tests.sh"""
+    return call_subprocess("./python/run-tests.sh", keyword_args={}, trail_args=args)
+
+
+def sbt(*args):
+    """Wrapper for build/sbt"""
+    required_env, _ = _get_required_env()
+    assert(not _)
```

Comment:
If you're going to assert on `_`, can we give it a name?

Reply:
fair enough, updated.